### PR TITLE
Remove one pod

### PR DESCRIPTION
**Pull request by Green-Star**
Should close issue #20.

I've just used the `quitFriends` method and adapt it to remove one friend.

I can see one problem with this method, though : It's not really DRY. Maybe we should add one method in the `server/lib/friends.js` to remove just one friend and call it 
- in the removeFriend method
- while iterating over the friend list in quitFriends 
?

### CODE REVIEW

File: server/lib/friends.ts [hunk 1]
```diff
@@ -229,6 +229,28 @@ function sendOwnedVideosToPod (podId: number) {
     })
 }
 
+function removeFriend (pod: PodInstance) {
+  // Stop pool requests
+  requestScheduler.deactivate()
```
Comment (Chocobozzz) [inline server/lib/friends.ts]: I'm not sure we need to stop the scheduler

File: server/tests/api/friends-advanced.js [hunk 2]
```diff
@@ -25,6 +25,27 @@ describe('Test advanced friends', function () {
...
 
+  function removeFriend (podNumber, podNumberToRemove, callback) {
+    const server = servers[podNumber - 1]
+    const serverToRemove = servers[podNumberToRemove - 1]
+
+    getFriendsList(podNumber, function (err, res) {
+      if (err) throw err
+
+      let friendsList = res.body.data
+      let podIdToRemove = -1
```
Comment (Chocobozzz) [inline server/tests/api/friends-advanced.js]: `podIdToRemove = friendsList.find(friend => friend.host === serverToRemove.host)`

File: server/tests/api/friends-basic.js [hunk 3]
```diff
@@ -198,6 +198,78 @@ describe('Test basic friends', function () {
...
+      // Pod 1 quits pod 2
+      function (next) {
+        const server = servers[0]
+
+        // Get pod 2 id so we can query it
+        podsUtils.getFriendsList(server.url, function (err, res) {
+          if (err) throw err
+
+          const result = res.body.data
+          let podId
```
Comment (Chocobozzz) [inline server/tests/api/friends-basic.js]: Same as above, use `.find`

File: server/tests/utils/pods.js [hunk 4]
```diff
@@ -90,6 +91,27 @@ function quitFriends (url, accessToken, expectedStatus, end) {
...
+  request(url)
+    .delete(path)
+    .set('Accept', 'application/json')
+    .set('Authorization', 'Bearer ' + accessToken)
+    .expect(expectedStatus)
+    .end(function (err, res) {
+      if (err) throw err
+
+      // Wait for the request between pods
+      setTimeout(end, 1000)
```
Comment (Chocobozzz) [inline server/tests/utils/pods.js]: `setTimeout` is not needed

File: server/lib/friends.ts [hunk 5]
```diff
@@ -242,6 +242,24 @@ function fetchRemotePreview (pod: PodInstance, video: VideoInstance) {
   return request.get(REMOTE_SCHEME.HTTP + '://' + host + path)
 }
 
+function removeFriend (pod: PodInstance) {
+  const requestParams = {
+    method: 'POST' as 'POST',
+    path: '/api/' + API_VERSION + '/remote/pods/remove',
+    sign: true,
```
Comment (Chocobozzz) [inline server/lib/friends.ts]: You don't need this anymore

File: server/lib/friends.ts [hunk 6]
```diff
@@ -242,6 +242,24 @@ function fetchRemotePreview (pod: PodInstance, video: VideoInstance) {
...
+    toPod: pod
+  }
+
+  return makeSecureRequest(requestParams)
+    .then(() => pod.destroy())
+    .then(() => {
+      logger.info('Removed friend.')
+    })
+    .catch(err => {
+      logger.error('Some errors while quitting friend %s (id: %d).', pod.host, pod.id, { err: err })
```
Comment (Chocobozzz) [inline server/lib/friends.ts]: `logger.error('Some errors while quitting friend %s (id: %d).', pod.host, pod.id, err)`

File: server/middlewares/validators/pods.ts [hunk 7]
```diff
@@ -58,9 +58,33 @@ function podsAddValidator (req: express.Request, res: express.Response, next: ex
...
+        if (!pod) {
+          logger.error('Cannot find pod %d.', req.params.id)
+          return res.sendStatus(404)
+        }
+
+        res.locals.pod = pod
+        return next()
+      })
+      .catch(err => {
+        logger.error('Cannot load pod %d.', req.params.id, { error: err })
```
Comment (Chocobozzz) [inline server/middlewares/validators/pods.ts]: `logger.error('Cannot load pod %d.', req.params.id, err)`

File: server/lib/friends.ts (hunk 5 above)
Comment (Green-Star) [inline server/lib/friends.ts]: We don't need to post a remove request to the pod we're quitting ?

Comment (Chocobozzz) [inline server/lib/friends.ts]: You don't need the `sign` parameter (it was removed recently, now all requests are automatically signed).

Comment (Green-Star) [inline server/lib/friends.ts]: Okay :+1:

**Pull request merged**